### ISSUE TITLE
DX12/GL/Vulkan: A-men 2. White textures on npc and interactive objects

### ISSUE THREAD

**Issue by O1L** [RPCS3/rpcs3#2170]
![1](https://cloud.githubusercontent.com/assets/5551509/18887349/5dcb5676-84fc-11e6-997e-0154b5feea5d.png)

**Comment by raven02** [comment]
Did DX12 and Vulkan render correct ?

**Comment by O1L** [comment]
My GeForce GT820m does not support DX12 and Vulkan, sorry.

**Comment by raven02** [comment]
I see. Thanks .

**Comment by raven02** [comment]
Vulkan looks good though hits shader error 

```
E {rsx::thread} RSX: Warning, version 420 is not yet complete; most version-specific features are present, but some are missing.
ERROR: 0:68: '' :  syntax error
ERROR: 1 compilation errors.  No code generated.


E {rsx::thread} RSX: 
F {rsx::thread} class std::runtime_error thrown: Failed to compile fragment shader
(in file C:\rpcs3\rpcs3\Emu\RSX\VK\VKFragmentProgram.cpp:475)

```

**Comment by raven02** [comment]
DX12 also stop at the same point with shader error

```
E {rsx::thread} RSX: Src type 3 used, please report this to a developer.
E {rsx::thread} RSX: FS build failed:0000000005118510
F {rsx::thread} class std::runtime_error thrown: fragment program compilation failure
(in file c:\rpcs3\rpcs3\emu\rsx\d3d12\D3D12PipelineState.h:162)
```

**Comment by SakataGintokiYT** [comment]
@raven02 
@AniLeo 
Fixed by #2405 on all API
